utilities/include/par.h

```diff
@@ -0,0 +1,156 @@
+    }                                                                          \
+    __builtin_unreachable();                                                   \
+  }
+#elif THRUST_DEVICE_SYSTEM == THRUST_DEVICE_SYSTEM_OMP
+#define THRUST_DYNAMIC_BACKEND_VOID(NAME, SUFFIX)                              \
+  template <typename... Args>                                                  \
+  void NAME##SUFFIX(ExecutionPolicy policy, Args... args) {                    \
+    switch (policy) {                                                          \
+    case ExecutionPolicy::ParUnseq:                                            \
+    case ExecutionPolicy::Par:                                                 \
+      thrust::NAME(thrust::omp::par, args...);                                 \
+      break;                                                                   \
+    case ExecutionPolicy::Seq:                                                 \
+      thrust::NAME(thrust::host, args...);                                     \
+      break;                                                                   \
+    }                                                                          \
+  }
+
+#define THRUST_DYNAMIC_BACKEND(NAME, RET)                                      \
+  template <typename Ret = RET, typename... Args>                              \
+  Ret NAME(ExecutionPolicy policy, Args... args) {                             \
+    switch (policy) {                                                          \
+    case ExecutionPolicy::ParUnseq:                                            \
+    case ExecutionPolicy::Par:                                                 \
+      return thrust::NAME(thrust::omp::par, args...);                          \
+    case ExecutionPolicy::Seq:                                                 \
+      return thrust::NAME(thrust::host, args...);                              \
+    }                                                                          \
+    __builtin_unreachable();                                                   \
+  }
+#else
+#define THRUST_DYNAMIC_BACKEND_VOID(NAME, SUFFIX)                              \
+  template <typename... Args>                                                  \
+  void NAME##SUFFIX(ExecutionPolicy policy, Args... args) {                    \
+    thrust::NAME(thrust::host, args...);                                       \
+  }
+
+#define THRUST_DYNAMIC_BACKEND(NAME, RET)                                      \
+  template <typename Ret = RET, typename... Args>                              \
+  Ret NAME(ExecutionPolicy policy, Args... args) {                             \
+    return thrust::NAME(thrust::host, args...);                                \
+  }
+#endif
+
+THRUST_DYNAMIC_BACKEND_VOID(gather, )
+THRUST_DYNAMIC_BACKEND_VOID(gather_if, )
+THRUST_DYNAMIC_BACKEND_VOID(remove_if, _void)
+THRUST_DYNAMIC_BACKEND_VOID(unique, _void)
```

Comment:
To avoid name collision with the one define below with `THRUST_DYNAMIC_BACKEND`. The reason for having these void variants is to avoid the need of specifying the return type via `remove_if<RET>` if we don't need it. I did this a few days ago but not sure if this is still needed, will have a look at that and remove these variants if they are not used.